### ISSUE TITLE
Number of steps for training

### ISSUE THREAD

**Issue by MounirB** [rekon/T3D-keras#2]
Why have you chosen to compute the number of steps like this : 
`
train_steps = len(d_train)//BATCH_SIZE
`
`
val_steps = len(d_valid)//BATCH_SIZE
`

**Comment by rekon** [comment]
Because, for each epoch, I want the model to train/validate through the entire training/validation data-set

**Comment by MounirB** [comment]
Is it possible to choose a lower number of steps (let's say 100) for each epoch ? What do you think about it ?

The current number of steps is at around 1200 considering the size the of the training set (related to UCF 50). With a 12Gigabytes GPU, one epoch can last for more than 2 hours.

**Comment by rekon** [comment]
Hi,

Yes it is absolutely fine to go for a less number of steps, but in that
case you will make a trade-off with the 'accuracy'. You can replace the
following statements -

> train_steps = len(d_train)//BATCH_SIZE
> val_steps = len(d_valid)//BATCH_SIZE

with

>  train_steps = min(len(d_train)//BATCH_SIZE,400)
> val_steps = min(len(d_valid)//BATCH_SIZE,100)

**Comment by MounirB** [comment]
Thank you for your answer ^^

**Comment by rekon** [comment]
Glad to help
